app/controllers/CategorisationPreparationController.scala

```diff
@@ -177,10 +177,15 @@ class CategorisationPreparationController @Inject() (
             record
           )
 
-          goodsRecordsConnector.updateCategoryAndComcodeForGoodsRecord(eori, recordId, record)
-        case Left(errors)  =>
-          val errorMessages = errors.toChain.toList.map(_.message).mkString(", ")
+          val result = for {
+            oldRecord <- goodsRecordsConnector.getRecord(eori, recordId)
+            _         <- goodsRecordsConnector.updateCategoryAndComcodeForGoodsRecord(eori, recordId, record, oldRecord)
```

Comment:
For this and other similar changes where we need to pass all the fields for a PUT, I have gone with making a call to retrieve the existing record to fill in the unchanged fields as opposed to picking them from the session. I did this following the pattern [here](https://github.com/hmrc/trader-goods-profiles-frontend/blob/25489227229e707a68a9523315437ded5f364168/app/controllers/CyaUpdateRecordController.scala#L241)